### ISSUE TITLE
Set previous charges to cancelled

### ISSUE THREAD

**Issue by paulcanning** [ohmybrew/laravel-shopify#79]
I'm at a point where I can let a user pick a new billing plan, which when processed, will set the old plans to cancelled on Shopify's end, but the record in the app DB is still saying 'active'.

The checks done for the last charge order by date, but it could be more robust.

To add to this, it'd be nice to have the BillingControllerTrait be a bit more generic. I need to be able to override stuff like how to deal with declined billing - currently it throws an exception (it's returning an abort, but I'm getting an exception page, instead of the normal abort page)

**Comment by ohmybrew** [comment]
I could mod it to provide a view instead?  Might help there.

For the charge, not sure I follow.. so you have an existing charge, you're upgrading/changing the plan, so you need the code to mark the current charge as 'cancelled' ? If so I can def look into that.

**Comment by paulcanning** [comment]
Yes, like how the AppUninstalledJob sets charges to cancelled when deleting the app.

I couldn't find a webhook that gives this information, so it's most probably the case you'd have to check previous charges when processing a charge.

**Comment by ohmybrew** [comment]
#4258693b3eb7430d426530faf1932cf70d1aa6cd Should fix this, will be in next release.
